static/js/src/advantage/subscribe/react/APICalls/Product.jsx

```diff
@@ -0,0 +1,37 @@
+import { useState } from "react";
+import { useQuery, useQueryClient } from "react-query";
+
+const useProduct = () => {
+  const [observerSet, setObserverSet] = useState(false);
+
+  const buyButton = document.querySelector("#buy-now-button");
```

Comment:
I assume in the future we will move to using `ref` but it would be better to target js-prefixed classes

Reply:
This is not idiomatic React. You'll want to modify the call signature of this hook to accept a ref to the component that you want to use here.

Reply:
In the future, hopefully, the product selector will share its state with the modal directly. I'll update it to use a ref and js- classes

Reply:
Actually, I remember now why I queried the button here. 6 components are subscribed to this query and I wanted to only search for this button once 
![image](https://user-images.githubusercontent.com/11927929/121859302-d35ce100-ccf7-11eb-9d0e-c960cc7190bf.png)
Can you think of a better pattern or should I just 
```
  const buyButton = document.querySelector(".js-buy-now-button");
```
in all of them and pass the button node to the function?


Also, looking into it I'm not sure it's possible to use refs here as the button I'm targeting lives outside of the react app. Or if it is I don't know how to do it

Reply:
Ahh ok, can you add some detailed comments to the code to outline this reasoning.